File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/mount/GameDataPaddockObjectListAddMessage.scala





















// Generated on 05/03/2013 10:34:41
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait GameDataPaddockObjectListAddMessage extends MessageDeserializer {
  val messageId = 5992

  protected trait Message extends BaseMessage with Serializable {
    val messageId = GameDataPaddockObjectListAddMessage.messageId

    val paddockItemDescription: Seq[PaddockItem.TypeImpl]

    def serialize(buf: Buffer) {
      buf.writeUShort(paddockItemDescription.length)
      for (entry <- paddockItemDescription) {
        entry.serialize(buf)
      }
    }

  }

  protected def create(paddockItemDescription: Seq[PaddockItem.TypeImpl]): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val paddockItemDescription = {
      val builder = collection.immutable.Seq.newBuilder[PaddockItem.TypeImpl]
      for (i <- (0 to buf.readUShort())) {
        builder += PaddockItem.deserialize(buf)
      }
      builder.result()
    }

    create(paddockItemDescription)
  }

}

object GameDataPaddockObjectListAddMessage extends GameDataPaddockObjectListAddMessage {
  case class MessageImpl(paddockItemDescription: Seq[PaddockItem.TypeImpl]) extends Message
  type Target = MessageImpl

  override def create(paddockItemDescription: Seq[PaddockItem.TypeImpl]) = MessageImpl(paddockItemDescription)

  def apply(paddockItemDescription: Seq[PaddockItem.TypeImpl]) = create(paddockItemDescription)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/GameDataPaddockObjectListAddMessage.java


// Generated on 05/08/2013 19:37:44
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class GameDataPaddockObjectListAddMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5992;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public PaddockItem[] paddockItemDescription;
    
    public GameDataPaddockObjectListAddMessage() { }
    
    public GameDataPaddockObjectListAddMessage(PaddockItem[] paddockItemDescription) {
        this.paddockItemDescription = paddockItemDescription;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeUShort(paddockItemDescription.length);
        for (PaddockItem entry : paddockItemDescription) {
            entry.serialize(buf);
        }
    }
    
    @Override
    public void deserialize(Buffer buf) {
        int limit = buf.readUShort();
        paddockItemDescription = new PaddockItem[limit];
        for (int i = 0; i < limit; i++) {
            paddockItemDescription[i] = new PaddockItem();
            paddockItemDescription[i].deserialize(buf);
        }
    }
    
}
